app/services/flowcommerce_spree/order_updater.rb

```diff
@@ -45,7 +45,7 @@ def finalize_order
     def complete_checkout
       upsert_data
       map_payments_to_spree
-      finalize_order
+      finalize_order unless @order.complete?
```

Comment:
```suggestion
      finalize_order if @order.state == 'complete'
```

because `@order.state = 'complete'` is set by `map_payments_to_spree` if the payments amounts sum is >= order.amount. If not, the state is not marked as `complete`.

Also, why not check by `@order.completed?` is because it is checking the `completed_at`, which will be set only in `finalize_order`.

Reply:
Ohh  I see your point. I misunderstood the intention of that validation. Thought it was to avoid calling finalize_order if the order was already completed. Makes sense!!